actions/ci-beholder-validator/README.md

```diff
@@ -0,0 +1,164 @@
+# Schema Evolution Validator
+
+A GitHub Action tool that validates schema evolution by checking compatibility
+with your schema registry when pull requests are created. The tool helps prevent
+breaking changes in your schemas by automatically validating them against the
+existing versions.
+
+## Overview
+
+When you create a pull request containing schema changes, this tool
+automatically:
+
+1. Reads schema definitions from your `beholder.yaml` configuration
+2. Makes compatibility checks against your schema registry
+3. Reports any compatibility issues in the GitHub Actions log
+
+This ensures that schema changes are caught early in the development cycle,
+before they can impact production services.
+
+## Getting Started
+
+### Prerequisites
+
+- A schema registry service with a compatibility check endpoint
+- GitHub repository with Actions enabled
+- Go 1.21 or later
+- Appropriate permissions to configure GitHub Secrets
+
+### Configuration
+
+Create a `beholder.yaml` file in your repository with your schema definitions:
+
+```yaml
+beholder:
+  domain: your_domain
+  schemas:
+    - entity: OnRampSend
+      schema: "./schemas/on_ramp_send.proto"
+    - entity: OffRampSend
+      schema: "./schemas/off_ramp_send.avsc"
+```
+
+Note: Schema paths should be relative to the `beholder.yaml` location.
+
+### A sample setup without using actions folder
+
+1. Create `.github/workflows/schema-check.yml`:
```

Comment:
Would it make sense to make a [reusable workflow](https://docs.github.com/en/actions/sharing-automations/reusing-workflows), which then calls this action? It would allow you to easier update the consumers of this action/workflow.

Both options work well though so no pressure.